src/chain/keysinterface.rs

```diff
@@ -109,43 +108,6 @@ impl_writeable!(ChannelKeys, 0, {
 	commitment_seed
 });
 
-impl ChannelKeys {
-	/// Generate a set of lightning keys needed to operate a channel by HKDF-expanding a given
-	/// random 32-byte seed
-	pub fn new_from_seed(seed: &[u8; 32]) -> ChannelKeys {
-		let mut prk = [0; 32];
-		hkdf_extract(Sha256::new(), b"rust-lightning key gen salt", seed, &mut prk);
```

Comment:
Oops, inverted pseudo-random key and info context ?

Reply:
No, should have extracted 32*N bytes instead of extracting 32 bytes N times with just a different salt. Technically thats not the way its supposed to work and doesn't mix as much but if it were actually ever a problem that would indicate a very serious issue with SHA256.